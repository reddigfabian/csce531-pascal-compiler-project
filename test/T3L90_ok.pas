(* Test file for level 100 of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L100ok;

var
   a,b : Integer;
   c   : Char;


procedure print_one; External;
procedure print_two; External;
procedure print_three; External;
procedure print_dunno; External;
procedure print_just; External;
procedure print_isnt; External;
procedure print_hello; External;
procedure print_oops; External;
procedure print_712; External;
procedure print_is_the; External;
procedure print_grooviest; External;
procedure print_newline; External;


begin

   a := 9;
   b := 3;

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   end; { case }

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   else
      print_dunno
   end; { case }

   a := 2;

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   else
      print_dunno
   end; { case }

   a := 1;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }

   a := 2;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }

   a := 3;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }


   { This next part is a lot easier with constant-folding. }

   c := chr(3);

   case c of
     chr(3) :
        begin
	   case c of
	     chr(1),chr(2),chr(4) : print_oops;
	     chr(3)		  : 
	        begin
		   print_712;
		   print_is_the
		end
	   end; { case }
	   print_grooviest
	end
   else	       
      print_oops;
      case ord(c) of
	1,2,3,4	: print_oops
      else	
	 print_oops
      end; { case }
   end; { case }
   
   print_newline

end.
