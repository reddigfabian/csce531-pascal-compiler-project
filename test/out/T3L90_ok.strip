b_global_decl (B, alignment = 4, size = 4)
b_global_decl (A, alignment = 4, size = 4)
b_global_decl (C, alignment = 1, size = 1)
b_func_prologue (main)
b_push_ext_addr (A)
b_push_const_int (9)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 1, .LC2, pop on jump )
b_jump ( destination = .LC2 )
b_jump ( destination = .LC1 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC0 )
b_dispatch (  == ,signed long int, 2, .LC5, pop on jump )
b_jump ( destination = .LC5 )
b_jump ( destination = .LC4 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_jump ( destination = .LC0 )
b_dispatch (  == ,signed long int, 3, .LC8, pop on jump )
b_jump ( destination = .LC8 )
b_jump ( destination = .LC7 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC0 )
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 1, .LC12, pop on jump )
b_jump ( destination = .LC12 )
b_jump ( destination = .LC11 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC10 )
b_dispatch (  == ,signed long int, 2, .LC15, pop on jump )
b_jump ( destination = .LC15 )
b_jump ( destination = .LC14 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_jump ( destination = .LC10 )
b_dispatch (  == ,signed long int, 3, .LC18, pop on jump )
b_jump ( destination = .LC18 )
b_jump ( destination = .LC17 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC10 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (2)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 1, .LC22, pop on jump )
b_jump ( destination = .LC22 )
b_jump ( destination = .LC21 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 2, .LC25, pop on jump )
b_jump ( destination = .LC25 )
b_jump ( destination = .LC24 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 3, .LC28, pop on jump )
b_jump ( destination = .LC28 )
b_jump ( destination = .LC27 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC20 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 2, .LC32, pop on jump )
b_jump ( destination = .LC32 )
b_dispatch (  == ,signed long int, 1, .LC32, pop on jump )
b_jump ( destination = .LC32 )
b_jump ( destination = .LC31 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC30 )
b_dispatch (  == ,signed long int, 3, .LC36, pop on jump )
b_jump ( destination = .LC36 )
b_jump ( destination = .LC35 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC30 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (2)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 2, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 1, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_jump ( destination = .LC39 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC38 )
b_dispatch (  == ,signed long int, 3, .LC44, pop on jump )
b_jump ( destination = .LC44 )
b_jump ( destination = .LC43 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC38 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 2, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_dispatch (  == ,signed long int, 1, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_jump ( destination = .LC47 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_jump ( destination = .LC46 )
b_dispatch (  == ,signed long int, 3, .LC52, pop on jump )
b_jump ( destination = .LC52 )
b_jump ( destination = .LC51 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC46 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (C)
b_push_const_int (3)
b_convert (signed long int -> unsigned char)
b_assign (unsigned char)
b_pop ()
b_push_ext_addr (C)
b_deref (unsigned char)
b_convert (unsigned char -> signed long int)
b_dispatch (  == ,signed long int, 3, .LC56, pop on jump )
b_jump ( destination = .LC56 )
b_jump ( destination = .LC55 )
b_push_ext_addr (C)
b_deref (unsigned char)
b_convert (unsigned char -> signed long int)
b_dispatch (  == ,signed long int, 4, .LC60, pop on jump )
b_jump ( destination = .LC60 )
b_dispatch (  == ,signed long int, 2, .LC60, pop on jump )
b_jump ( destination = .LC60 )
b_dispatch (  == ,signed long int, 1, .LC60, pop on jump )
b_jump ( destination = .LC60 )
b_jump ( destination = .LC59 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC58 )
b_dispatch (  == ,signed long int, 3, .LC65, pop on jump )
b_jump ( destination = .LC65 )
b_jump ( destination = .LC64 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_712, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_is_the, void)
b_jump ( destination = .LC58 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_grooviest, void)
b_jump ( destination = .LC54 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_push_ext_addr (C)
b_deref (unsigned char)
b_convert (unsigned char -> signed long int)
b_dispatch (  == ,signed long int, 4, .LC69, pop on jump )
b_jump ( destination = .LC69 )
b_dispatch (  == ,signed long int, 3, .LC69, pop on jump )
b_jump ( destination = .LC69 )
b_dispatch (  == ,signed long int, 2, .LC69, pop on jump )
b_jump ( destination = .LC69 )
b_dispatch (  == ,signed long int, 1, .LC69, pop on jump )
b_jump ( destination = .LC69 )
b_jump ( destination = .LC68 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC67 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_func_epilogue (main)
