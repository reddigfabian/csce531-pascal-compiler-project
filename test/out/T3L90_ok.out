dunno what a is
a is two
a is one
a is one
a is three
712 is the grooviest...
